### PR TITLE
fix(decimalZero): correct false positive

### LINKED ISSUE #137: False Positive with "decimalZero"
Given `decimalZero: "leading"` and the following Less code...

```less
.foobar {
  color: rgba(255, 255, 255, .5);
}
```

I should get an error on line 2, because `.5` should be `0.5`

### PR DESCRIPTION
**Pull request by CITguy**
* check all "number" values in declaration
* more robust regexes
* refactored tests

closes #137

### CODE REVIEW

File: lib/linters/decimal_zero.js [hunk 1]
```diff
@@ -9,58 +9,67 @@ module.exports = {
...
-            inclusion: 'with',
-            type: null
-        };
+        var result = [];
+        var self = this;
 
-        node.forEach('value', function (element) {
-            value = element.first('dimension');
-            value = value && value.first('number');
-        });
```
Comment (CITguy) [inline lib/linters/decimal_zero.js]: This logic would iterate over all "value" nodes, but the `value` variable would only be set to the last one in the iteration.

File: lib/linters/decimal_zero.js [hunk 2]
```diff
@@ -9,58 +9,67 @@ module.exports = {
...
+        node.traverseByType('number', function (element) {
```
Comment (CITguy) [inline lib/linters/decimal_zero.js]: Wrapped linting logic with `traverseByType()` to run linting on *each* number found in declaration.

**Comment by jwilsson** [comment]
LGTM :+1:

**Comment by CITguy** [comment]
Is AppVeyor having issues? I don't think I have access to see the lesshint project on appveyor.

**Comment by CITguy** [comment]
reopened to re-trigger app integrations

**Comment by jwilsson** [comment]
I'm gonna delete the project from AppVeyor and add it again. It's pointing to the old project (jwilsson/lesshint) so that might cause some issues.

**Comment by jwilsson** [comment]
Reopened to try and trigger new build. Sorry for spamming notifications.

**Comment by CITguy** [comment]
no problem :)

**Comment by jwilsson** [comment]
I can't seem to trigger a AppVeyor build of this but Travis passes and it seems unlikely that AppVeyor will fail, and if it does, we'll catch it anyway.

I'm good with merging unless someone else thinks different? :)

**Comment by CITguy** [comment]
Travis tested it 7 times, and passed each time. I'm good with merging, if you are.

**Pull request merged**